packages/common/src/projects/types.ts

```diff
@@ -0,0 +1,199 @@
+import { IUserRequestOptions } from "@esri/arcgis-rest-auth";
+import { IUserItemOptions } from "@esri/arcgis-rest-portal";
+import { IRequestOptions } from "@esri/arcgis-rest-request";
+import { IGeometry, IPoint } from "@esri/arcgis-rest-types";
+
+export type IHubProjectOptions = Pick<Partial<IHubProject>, "title" | "org">;
+
+/**
+ * Base properties for Hub Models
+ * This is a subset of IItem, that can apply to
+ * models that are not backed by items
+ */
+export interface IHubBaseModel {
+  readonly id?: string;
+  readonly owner?: string;
+  title: string;
+  description?: string;
+  snippet?: string;
+  extent?: number[][];
+  culture?: string;
+  properties?: any;
+  url?: string;
+  tags?: string[];
+  typeKeywords?: string[];
+  readonly created?: number;
+  readonly modified?: number;
+}
+
+/**
+ * Defines the properties of a Hub Project object
+ */
+export interface IHubProject extends IHubBaseModel, IWithSlug, IWithLayout {
+  timeline?: IHubTimeline;
+}
+
+export interface IWithSlug {
+  slug?: string;
+  org: { id: string; key: string };
+}
+
+export interface IWithLayout {
+  layout?: IHubLayout;
+}
+
+/**
+ * Defined layout related functions that a module
+ * working with IWithLayout objects must implement
+ */
+export interface IWithLayoutFns {
```

Comment:
Since the POJO's don't have fn's (not classes) I'm thinking we can separate the properties from the fns, and then the Hub<Type> classes can implement these `IWith<Behavior>Fns` interfaces. Muddier than pure OOP, but I think this will get us a lot of benefits, and still allow this all to play nice w/ web frameworks that ❤️  immutable pojos